src/midiEditorCommands.cpp

```diff
@@ -50,6 +50,29 @@ int getItemPPQ(MediaItem* item) {
 	return ppq;
 }
 
+// return the midi editor zoom ratio of the item 
+double getMidiZoomRatio(MediaItem* item) {
+	static const regex re("CFGEDITVIEW -?[0-9.]+ ([0-9.]+) ");
+	const size_t startBuffSize = 16384;
+	const size_t maxBuffSize = 1<<24; // 16 Mib
+	for (size_t buffSize = startBuffSize; buffSize <= maxBuffSize; buffSize *= 2) {
+		string buffer(buffSize, '\0');
+		if (!GetItemStateChunk(item, &buffer.front(), buffSize, false)) {
+			return -1;
+		}
+		smatch match;
+		if (!regex_search(buffer, match, re)) {
```

Comment:
This is super clever. That said, I worry about performance here. We had to disable solo defeat reporting back when it used a regexp to search a state chunk; see #383. Of course, this isn't as bad because it will only impact the user if they actually need to have the zoom level reported. On the other hand, the user is more likely to press it many times in rapid succession and I worry this will cause major lag on large MIDI items if the user pressed it many times rapidly.

I guess we'll just have to see how it goes in practice. I agree having this is better than nothing, but I really wish Cockos would just expose this.

Reply:
> I agree having this is better than nothing, but I really wish Cockos would just expose this.
> 
Shall I try Cockos again? I'm thinking perhaps Justin might juggle it up the to-do list if I show him that Robbie's done some wizardry but the best way we've got available is still gonna hit a wall performance-wise.

Reply:
Let's see if users actually complain. If they don't, no point in ringing the bell again. If they do, well, it's more likely to get fixed if we can show it's hurting users.

That said, this take problem Robbie has discovered might be a show-stopper and require us to push on it anyway.